### PR TITLE
fixes #1150 and #1191

### LINKED ISSUE #1150: Login error: JSON decode fail, 403 - Forbidden.
**Describe the bug**
Today I tried to login from two different accounts, from different IPs, and received a login error, 429 forbidden.

**To Reproduce**
`instaloader --login <username> profile nasa`

**Expected behavior**
I expected to login and start downloading posts.

**Error messages and tracebacks**
Using above command line, I receive this:
`Fatal error: Login error: JSON decode fail, 403 - Forbidden.`

Using module, I get:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/instaloader/instaloadercontext.py", line 224, in login
    resp_json = login.json()
  File "/usr/local/lib/python3.7/dist-packages/requests/models.py", line 900, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.7/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/ig-login.py", line 57, in <module>
    L.login(USER, PASSWORD)
  File "/usr/local/lib/python3.7/dist-packages/instaloader/instaloader.py", line 546, in login
    self.context.login(user, passwd)
  File "/usr/local/lib/python3.7/dist-packages/instaloader/instaloadercontext.py", line 228, in login
    ) from err
instaloader.exceptions.ConnectionException: Login error: JSON decode fail, 403 - Forbidden.
```

**Instaloader version**
4.7.1

**Additional context**
If I try to login via web or via app, I don't have any problems. My accounts seem perfectly fine. Also, without logging in, e.g.
`instaloader nasa`
I can download posts, so I guess my IP range is not banned.

### PR DESCRIPTION
**Pull request by twitter-79**
taking csrf token from the instagram login page as it is empty using the old url

this should fix JSONDecodeError when trying to log in (at least in Europe). The error was described in #1150 as well as #1191

### CODE REVIEW

File: instaloader/instaloadercontext.py [hunk 1]
```diff
@@ -210,8 +210,8 @@ def login(self, user, passwd):
         # Override default timeout behavior.
         # Need to silence mypy bug for this. See: https://github.com/python/mypy/issues/2427
         session.request = partial(session.request, timeout=self.request_timeout) # type: ignore
-        session.get('https://www.instagram.com/web/__mid/')
-        csrf_token = session.cookies.get_dict()['csrftoken']
+        res = session.get('https://www.instagram.com/accounts/login/')
+        csrf_token = re.findall(r"csrf_token\":\"(.*?)\"", res.text)[0]
```
Comment (aandergr) [inline instaloader/instaloadercontext.py:214]:
```suggestion
        csrf_json = self.get_json('accounts/login/', {}, session=session)
        csrf_token = csrf_json['config']['csrf_token']
```

**Review by aandergr — COMMENTED** [review]
Good call!

I recommend to simply use `get_json` for the JSON parsing instead of a new regex magic and have added a inline comment for that.

File: instaloader/instaloadercontext.py (hunk 1 above)
Comment (twitter-79) [reply instaloader/instaloadercontext.py:214]: just tried out this suggestion, works fine for me as well!

this is my first pull request, so no clue on what to do now (gitlab user normally)? Feel free to recommend the next steps :-D

**Comment by aandergr** [comment]
Looks good – thanks a lot!

**Pull request merged**